datastream-kafka-connector/src/test/java/com/linkedin/datastream/connectors/kafka/mirrormaker/TestKafkaMirrorMakerConnectorTask.java

```diff
@@ -366,6 +367,44 @@ boolean isPostShutdownHookExceptionCaught() {
     }
   }
 
+  @Test
+  public void testPartitionManagedLockAcquireFailMetric() throws InterruptedException {
```

Comment:
Please feel free to ignore this comment (and excuse its length).

One thing I personally found unnerving about adding/fixing metrics in the past is that one needs to make sure the metrics created/registered through `DynamicMetricsManager` have to match the `BrooklinMetricInfo` list returned by `getMetricInfos()`. 

For instance:
1. I had to make sure I returned the correct sub-type of `BrooklinMetricInfo` from `getMetricInfos()` (e.g. `BrooklinMeterInfo` for `Meter`s, `BrooklinCounterInfo` for `Counter`s ... etc). Otherwise, I'd be creating the wrong InGraph (or maybe it won't work at all, who knows)

2. I had to make sure every metric I created through `DynamicMetricsManager` matched one — and only one — of the `BrooklinMetricInfo` objects returned by `getMetricInfos()` by name (or regex). If a metric were to match more than one `BrooklinMetricInfo`, multiple InGraphs would be created for it. If I were to get the regex wrong, no `BrooklinMetricInfo` would match my metric and no InGraph would be created for it.

To address these concerns, I wrote `MetricsTestUtils` which exposes the API below for asserting the above:
```java
  public static void verifyMetrics(MetricsAware metricsAware, DynamicMetricsManager metricsManager)
```
If you're interested in using this utility, you can do so by adding the following snippet to your test:
```java
    MetricsTestUtils.verifyMetrics(new MetricsAware() {
      @Override
      public List<BrooklinMetricInfo> getMetricInfos() {
        return KafkaMirrorMakerConnectorTaskTest.getMetricInfos("");
      }
    }, DynamicMetricsManager.getInstance());
```
I tried it myself and the test is passing but I'm just sharing some past thoughts I had about testing metrics in general.

Reply:
Thank you so much for this suggestion! Completely forgot about this utility. I've added it, and I think this should be advertised more. Might help us catch some of those mbean bugs where we create duplicate metrics.

Reply:
My pleasure. I'm putting together a developer's guide for working with metrics. I'll make sure to mention this utility. Thank you.